include/kalmar_math.h

```diff
@@ -10,286 +10,288 @@
-extern "C" double __hc_fma_double(double x, double y, double z) restrict(amp);
-
-extern "C" __fp16 __hc_fmax_half(__fp16 x, __fp16 y) restrict(amp);
-extern "C" float __hc_fmax(float x, float y) restrict(amp);
-extern "C" double __hc_fmax_double(double x, double y) restrict(amp);
-
-extern "C" __fp16 __hc_fmin_half(__fp16 x, __fp16 y) restrict(amp);
-extern "C" float __hc_fmin(float x, float y) restrict(amp);
-extern "C" double __hc_fmin_double(double x, double y) restrict(amp);
-
-extern "C" __fp16 __hc_fmod_half(__fp16 x, __fp16 y) restrict(amp);
-extern "C" float __hc_fmod(float x, float y) restrict(amp);
-extern "C" double __hc_fmod_double(double x, double y) restrict(amp);
-
-extern "C" int __hc_fpclassify_half(__fp16 x) restrict(amp);
-extern "C" int __hc_fpclassify(float x) restrict(amp);
-extern "C" int __hc_fpclassify_double(double x) restrict(amp);
-
-extern "C" __fp16 __hc_frexp_half(__fp16 x, int *exp) restrict(amp);
-extern "C" float __hc_frexp(float x, int *exp) restrict(amp);
-extern "C" double __hc_frexp_double(double x, int *exp) restrict(amp);
-
-extern "C" __fp16 __hc_hypot_half(__fp16 x, __fp16 y) restrict(amp);
-extern "C" float __hc_hypot(float x, float y) restrict(amp);
-extern "C" double __hc_hypot_double(double x, double y) restrict(amp);
-
-extern "C" int __hc_ilogb_half(__fp16 x) restrict(amp);
-extern "C" int __hc_ilogb(float x) restrict(amp);
-extern "C" int __hc_ilogb_double(double x) restrict(amp);
-
-extern "C" int __hc_isfinite_half(__fp16 x) restrict(amp);
-extern "C" int __hc_isfinite(float x) restrict(amp);
-extern "C" int __hc_isfinite_double(double x) restrict(amp);
-
-extern "C" int __hc_isinf_half(__fp16 x) restrict(amp);
-extern "C" int __hc_isinf(float x) restrict(amp);
-extern "C" int __hc_isinf_double(double x) restrict(amp);
-
-extern "C" int __hc_isnan_half(__fp16 x) restrict(amp);
-extern "C" int __hc_isnan(float x) restrict(amp);
-extern "C" int __hc_isnan_double(double x) restrict(amp);
-
-extern "C" int __hc_isnormal_half(__fp16 x) restrict(amp);
-extern "C" int __hc_isnormal(float x) restrict(amp);
-extern "C" int __hc_isnormal_double(double x) restrict(amp);
-
-extern "C" __fp16 __hc_ldexp_half(__fp16 x, std::int16_t exp) [[hc]];
-extern "C" float __hc_ldexp(float x, int exp) restrict(amp);
-extern "C" double __hc_ldexp_double(double x, int exp) restrict(amp);
-
-extern "C" __fp16 __hc_lgamma_half(__fp16 x) restrict(amp);
-extern "C" float __hc_lgamma(float x) restrict(amp);
-extern "C" double __hc_lgamma_double(double x) restrict(amp);
-
-extern "C" __fp16 __hc_log_half(__fp16 x) restrict(amp);
-extern "C" float __hc_log(float x) restrict(amp);
-extern "C" float __hc_log_native(float x) restrict(amp);
-extern "C" double __hc_log_double(double x) restrict(amp);
 
-extern "C" __fp16 __hc_log10_half(__fp16 x) restrict(amp);
-extern "C" float __hc_log10(float x) restrict(amp);
-extern "C" double __hc_log10_double(double x) restrict(amp);
+extern "C" __fp16 __ocml_acos_half(__fp16 x) [[hc]];
```

Comment:
The correct OCML suffix is _f16 not _half.  Also should hcc be using __fp16 directly here?  Isn't there a better typedef like "half"?

Reply:
Actually this should use _Float16, which is the Clang kosher 16-bit float type. `typedef` half is awful though because there are a great many uses of that word, that need not have anything to do with a type (e.g. a function that halves). It's unlikely the C++ Standard will take `half` as the type name precisely for that reason (it's prime real-estate). Boost tries to emulate `<cstdint>` for FP here: https://www.boost.org/doc/libs/1_65_1/boost/math/cstdfloat/cstdfloat_types.hpp. But _Float16 is nice and explicative and also avoids the trap of looking too cute.

Reply:
`make test` didn't catch the problem so looks like there's a gap in the unit tests.  Could you add unit tests for these?

Reply:
ok